runtime-modules/argo-bridge/src/tests/mod.rs

```diff
@@ -0,0 +1,370 @@
+            remote_chains: Some(remote_chains),
+        };
+        assert_ok!(ArgoBridge::update_bridge_constrains(
+            RuntimeOrigin::root(),
+            parameters
+        ));
+
+        let transfer_amount = joy!(1000);
+        assert_ok!(Balances::set_balance(
+            RuntimeOrigin::root(),
+            account!(1),
+            transfer_amount + fee - 1,
+            joy!(0)
+        ));
+
+        let remote_account = RemoteAccount {
+            account: [0; 32],
+            chain_id: 1,
+        };
+        let result = ArgoBridge::request_outbound_transfer(
+            RuntimeOrigin::signed(account!(1)),
+            remote_account,
+            transfer_amount,
+            fee,
+        );
+        assert_err!(result, Error::<Test>::InsufficientJoyBalance);
+    });
+}
+
+#[test]
+fn request_outbound_transfer_with_unexpected_fee() {
+    with_test_externalities(|| {
+        let remote_chains = BoundedVec::try_from(vec![1u32]).unwrap();
+        let parameters = BridgeConstraints {
+            operator_account: Some(account!(1)),
+            pauser_accounts: Some(vec![account!(2), account!(3)]),
+            bridging_fee: Some(joy!(20)),
+            thawn_duration: None,
+            remote_chains: Some(remote_chains),
+        };
+        assert_ok!(ArgoBridge::update_bridge_constrains(
+            RuntimeOrigin::root(),
+            parameters
+        ));
+
+        let transfer_amount = joy!(1000);
+        assert_ok!(Balances::set_balance(
+            RuntimeOrigin::root(),
+            account!(1),
+            transfer_amount + 50,
+            joy!(0)
+        ));
+
+        let remote_account = RemoteAccount {
+            account: [0; 32],
+            chain_id: 1,
+        };
+        let result = ArgoBridge::request_outbound_transfer(
+            RuntimeOrigin::signed(account!(1)),
+            remote_account,
+            transfer_amount,
+            joy!(10),
+        );
+        assert_err!(result, Error::<Test>::FeeDifferentThanExpected);
+    });
+}
+
+#[test]
+fn finalize_inbound_transfer_success() {
+    with_test_externalities_custom_mint_allowance(joy!(1000), || {
+        let remote_chains = BoundedVec::try_from(vec![1u32]).unwrap();
+        let parameters = BridgeConstraints {
+            operator_account: Some(account!(1)),
+            pauser_accounts: None,
+            bridging_fee: None,
+            thawn_duration: None,
+            remote_chains: Some(remote_chains),
+        };
+        assert_ok!(ArgoBridge::update_bridge_constrains(
+            RuntimeOrigin::root(),
+            parameters
+        ));
+
+        let remote_transfer = RemoteTransfer { id: 0, chain_id: 1 };
+        let result = ArgoBridge::finalize_inbound_transfer(
+            RuntimeOrigin::signed(account!(1)),
+            remote_transfer,
+            account!(2),
+            1000,
+        );
+        assert_err!(result, Error::<Test>::InsufficienBridgMintAllowance);
```

Comment:
Name of this test ends in `_success` but we assert error here. I think this should be the happy case? In that case, same questions as for the outbound transfer happy case - can we assert change in mint allowance and in token balances?

Reply:
> Name of this test ends in `_success` but we assert error here. I think this should be the happy case? In that case, same questions as for the outbound transfer happy case - can we assert change in mint allowance and in token balances?

the title was good but the test was wrong...
added checks for the balance and the event deposit